skin_meta_api.lua

```diff
@@ -174,6 +201,14 @@ function skin_class:apply_skin_to_player(player)
 			y = self:get_meta("visual_size_y") or 1
 		}
 	})
+
+	local hand = self:get_hand()
+	if hand then
+		player:get_inventory():set_size("hand", 1)
+		player:get_inventory():set_stack("hand", 1, hand)
+	else
+		player:get_inventory():set_stack("hand", 1, "")
```

Comment:
To ensure sanity, this list needs to be "write"-protected using `minetest.register_allow_player_inventory_action`. Even if this list is not exposed by a GUI, CSM mods can allow access to it.

Reply:
Do I understand the minetest.register_allow_player_inventory_action() right  I just need to return 0? How I can test it?

Reply:
> How I can test it?

Add the slot to any formspec `list[current_player;hand;XX,YY;1,1;]`. It does not matter which one.

```lua
minetest.register_allow_player_inventory_action(function(player, action, inv, data)
	if data.to_list == "hand" or data.from_list == "hand" or data.listname == "hand" then
		return 1
	end
end)
```

Reply:
Added the function into init.lua. Tested shortly without and with the code. Work as expected.